### PR TITLE
Upgrade ASM to latest

### PR DESCRIPTION
**Pull request by basil**
Might fix the recent errors in https://ci.jenkins.io/job/Reporting/job/deprecated-usage-in-plugins ?

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]
> This resolves the problem for me locally.

Same here.

**Comment by basil** [comment]
A typical scenario for a contributor to the Jenkins project: two approvals, including one from someone with write access to the repository, no outstanding requests for change, and yet no merge.

**Comment by NotMyFault** [comment]
cc @timja or whoever monitors this repository 👀

**Comment by basil** [comment]
I fail to understand why the existing approver with write access did not merge the PR.

**Comment by basil** [comment]
BTW it may be that ASM becomes unnecessary in the future if [this JEP](https://openjdk.org/jeps/8280389) is completed.

**Pull request merged**
